TSOClient/tso.simantics/NetPlay/EODs/Handlers/VMEODSlotsPlugin.cs

```diff
@@ -0,0 +1,678 @@
+                        {
+                            if (slotsStops[betAndSlotResults[3]].Equals(VMEODSlotsStops.Second))
+                            {
+                                winnings = betAmount * TWO_TWO_TWO_PAYOUT_MULTIPLIER;
+                            }
+                        }
+                        break;
+                    }
+                case (byte)VMEODSlotsStops.First:
+                    {
+                        if (slotsStops[betAndSlotResults[2]].Equals(VMEODSlotsStops.First))
+                            winnings = betAmount * ONE_ONE_ANY_PAYOUT_MULTIPLIER;
+                        else
+                            winnings = betAmount * ONE_ANY_ANY_PAYOUT_MULTIPLIER;
+                        break;
+                    }
+                default: // losing roll, winnings remain 0
+                    break;
+            }
+            if (winnings > 0)
+            {
+                // dispatch event to tell object the winning amount
+                userClient.SendOBJEvent(new VMEODEvent((short)VMOEDSlotsObjectEvents.SetWinAmount, new short[1] { Convert.ToInt16(winnings) }));
+                // dispatch event to tell the object that the user has won
+                userClient.SendOBJEvent(new VMEODEvent((short)VMOEDSlotsObjectEvents.ExecuteWin, new short[1] { betAmount }));
+            }
+            else
+            {
+                // dispatch event to tell object that the user has lost
+                userClient.SendOBJEvent(new VMEODEvent((short)VMOEDSlotsObjectEvents.ExecuteLoss, new short[1] { betAmount }));
+            }
+
+            // send winnings and spins results to UI
+            userClient.Send("slots_spin", new Byte[] { Convert.ToByte(winnings / 255), Convert.ToByte(winnings % 255),
+                (byte)slotsStops[betAndSlotResults[1]], (byte)slotsStops[betAndSlotResults[2]], (byte)slotsStops[betAndSlotResults[3]] });
+            State = VMEODSlotsStates.Running;
+        }
+        private void SlotsCloseHandler(string evt, string arg, VMEODClient client)
+        {
+            Server.Disconnect(client);
+        }
+        private int[] RollNewGame(int betAmount)
+        {
+            int wheelOneStop = slotsRandom.Next(totalStops);
+            int wheelTwoStop = slotsRandom.Next(totalStops);
+            int wheelThreeStop = slotsRandom.Next(totalStops);
+            return new int[4] { betAmount, wheelOneStop, wheelTwoStop, wheelThreeStop};
+        }
+        private void WheelsStoppedHandler(string evt, string payoutString, VMEODClient client)
+        {
+            if (payoutString[0] != '0')
+            {
+                State = VMEODSlotsStates.DisplayingWin;
+                // send win event with random number for message displayed
+                client.Send("slots_display_win", "" + slotsRandom.Next(25, 30) + "%" + payoutString);
+
+                // pay the player
+                short winnings = Int16.Parse(payoutString);
```

Comment:
Did you just get the payout value... From the client? Then use it to give someone money? I cannot accept this pull request if something like this is possible.